Subject: Linux-RT 2.6.25-rt
From: http://www.kernel.org/pub/linux/kernel/projects/rt/
Acked-by: Sven-Thorsten Dietrich <sdietrich@suse.de>
From: Steven Rostedt <srostedt@redhat.com>
Subject: Make the task State char-string visible to all

The tracer wants to be able to convert the state number
into a user visible character. This patch pulls that conversion
string out the scheduler into the header. This way if it were to
ever change, other parts of the kernel will know.

Signed-off-by: Steven Rostedt <srostedt@redhat.com>
---
 include/linux/sched.h |    2 ++
 kernel/sched.c        |    2 +-
 2 files changed, 3 insertions(+), 1 deletion(-)

Index: linux-2.6.24-rt2/include/linux/sched.h
===================================================================
--- linux-2.6.24-rt2.orig/include/linux/sched.h	2008-01-30 16:43:33.000000000 -0500
+++ linux-2.6.24-rt2/include/linux/sched.h	2008-01-30 16:43:37.000000000 -0500
@@ -2003,6 +2003,8 @@ static inline void migration_init(void)
 }
 #endif
 
+#define TASK_STATE_TO_CHAR_STR "RSDTtZX"
+
 #endif /* __KERNEL__ */
 
 #endif
Index: linux-2.6.24-rt2/kernel/sched.c
===================================================================
--- linux-2.6.24-rt2.orig/kernel/sched.c	2008-01-30 16:43:37.000000000 -0500
+++ linux-2.6.24-rt2/kernel/sched.c	2008-01-30 16:43:37.000000000 -0500
@@ -4834,7 +4834,7 @@ out_unlock:
 	return retval;
 }
 
-static const char stat_nam[] = "RSDTtZX";
+static const char stat_nam[] = TASK_STATE_TO_CHAR_STR;
 
 static void show_task(struct task_struct *p)
 {
